### PR TITLE
Cal 67 - Modify Chipping Service to Chip by Pixel Instead of Lon/Lat.

### PR DESCRIPTION
**Pull request by dcruver**
#### What does this PR do?
This changes adds a method to the ChipService to chip by pixel.  There are also fixes to the original chip method implementation and changes to the Chip MetacardTransformer so that it now uses the chip-by-pixel method.

#### Who is reviewing it?
@jlcsmith 
@jaymcnallie 
@spearskw 

#### How should this be tested?
1) ingest a NITF
2) request a chip from the metacard transformer at https://localhost:8993/services/catalog/sources/ddf.distribution/{{resource-id}}?transform=chip&x={{top-left-corner-x}}&y={{top-left-corner-y}}&w={{chip-width}}&h={{chip-height}.
  --verify that the returned image is the size requested and appears to be the right area.

#### What are the relevant tickets?
  CAL-67

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/imaging/catalog-imaging-service-impl/src/test/java/org/codice/alliance/imaging/chip/service/impl/Boundary.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    private Vector scaleByLatitude(Vector v, double latitude) {
+        double lat = v.get(1);
+        double lon = v.get(0) / Math.cos(Math.toRadians(latitude));
+        return new BasicVector(new double[] {lon, lat});
+    }
+
+    private void createBoundary(double lat, double lon, double sizeInKm, double rotation) {
+        boundary = getUnitSquare().stream()
+                .map(v -> v.multiply(sizeInKm / 111))
```
Comment (bdeining) [inline catalog/imaging/catalog-imaging-service-impl/src/test/java/org/codice/alliance/imaging/chip/service/impl/Boundary.java]: What is 111?

Comment (spearskw) [inline catalog/imaging/catalog-imaging-service-impl/src/test/java/org/codice/alliance/imaging/chip/service/impl/Boundary.java]: 1 degree of longitude and latitude at the equator is roughly equal to 111 kilometers, so this is how you convert from kilometers to degrees. Would probably be helpful to assign that value to a constant with a good name.

File: catalog/imaging/catalog-imaging-transformer-chip/src/main/java/org/codice/alliance/imaging/chip/transformer/CropAdapter.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+        int x = getArg(args, "x");
+        int y = getArg(args, "y");
+        int w = getArg(args, "w");
+        int h = getArg(args, "h");
+
+        double scaleFactor = 1.0;
+
+        if (original != null && overview != null) {
+            scaleFactor = original.getWidth() / overview.getWidth();
```
Comment (bdeining) [inline catalog/imaging/catalog-imaging-transformer-chip/src/main/java/org/codice/alliance/imaging/chip/transformer/CropAdapter.java]: What about images that are not square?  The scaleFactor for y will be different than the scaleFactor for x

Comment (dcruver) [inline catalog/imaging/catalog-imaging-transformer-chip/src/main/java/org/codice/alliance/imaging/chip/transformer/CropAdapter.java]: No, the amount that we scale on both sides is the same as a percentage of their overall length.  Differing x and y scale factors would result in an aspect ratio that differs from the original image.  So if the original image is 2048x1024 and we scale it down to a max of 1024, then we're scaling both sides by .5 even though it's not a square image.

**Comment by bdeining** [comment]
LGTM after rebase.  Verified that the chipping works accurately on a scaled image by using the front end in https://github.com/codice/alliance/pull/62

**Comment by jlcsmith** [comment]
👍

**Comment by bdeining** [comment]
Built and tested after rebase 👍

**Comment by jaymcnallie** [comment]
👍

**Pull request merged**